### ISSUE TITLE
long_description_content_type flag in setup.py is ignored. IE #4079 not fixed.

### ISSUE THREAD

**Issue by hlovatt** [pypa/warehouse#9327]
The long_description_content_type flag in `setup.py` is still ignored despite original bug #4079 now closed and marked as fixed.

For my `setup.py` of:

    long_description=read_text('README.md'),
    long_description_content_type='text/markdown',

I get the following errors:

    `long_description` has syntax errors in markup and would not be rendered on PyPI.
    line 112: Warning: Inline literal start-string without end-string.
    warning: `long_description_content_type` missing. defaulting to `text/x-rst`.

The syntax error is not correct, the README renders on everything I have tried!

Therefore suspect that `long_description_content_type` error, which isn't missing, is the problem!

All files at:

   https://github.com/hlovatt/statmach

**Comment by di** [comment]
Hi, can you attach the built distributions you're attempting to upload?

**Comment by ewjoachim** [comment]
TL;DR:
- The root cause: https://github.com/pypa/setuptools/issues/1390
- A similar problem: https://github.com/pypa/warehouse/issues/9315
- Your fix:

Replace `license=read_text('LICENSE')` with `license="MIT License"`

---

Hello,

I've tried your files at [hlovatt/statmach](https://github.com/hlovatt/statmach). More precisely, I've tried the following command:
```
python setup.py sdist && twine check --strict dist/*
```

As you'll notice, this command doesn't involve PyPI: it builds the package, and then checks it using Twine (nothing is sent).

Even if the current state of your master branch replaces the long description:
```
long_description=f'See {statmach.__repository__} for detailed description.\n',  # read_text('README.md'),
```
it still seems to fail with:
```
Checking dist/statmach-1.0.8.tar.gz: FAILED, due to warnings
  warning: `long_description_content_type` missing. defaulting to `text/x-rst`.
```
The problem is that you're loading the entire license file in the `license` field rather than just `MIT License`. Multiline strings create [problems](https://github.com/pypa/setuptools/issues/1390
) with setuptools as seen in #9315.

The fix is to only let `long_description` contain newlines. This is not #4079.

**Comment by hlovatt** [comment]
Many thanks for the help. Multiline license text is indeed the problem.
